templates/contract-deploy/run-contract-setup.sh

```diff
@@ -0,0 +1,185 @@
+sed -i 's#http://127.0.0.1:8545#{{.l1_rpc_url}}#' hardhat.config.ts
+
+# Deploy gas token.
+# shellcheck disable=SC1054,SC1083
+{{if .zkevm_use_gas_token_contract}}
+echo_ts "Deploying gas token to L1"
+printf "[profile.default]\nsrc = 'contracts'\nout = 'out'\nlibs = ['node_modules']\n" > foundry.toml
+forge create \
+    --json \
+    --rpc-url "{{.l1_rpc_url}}" \
+    --mnemonic "{{.l1_preallocated_mnemonic}}" \
+    contracts/mocks/ERC20PermitMock.sol:ERC20PermitMock \
+    --constructor-args  "CDK Gas Token" "CDK" "{{.zkevm_l2_admin_address}}" "1000000000000000000000000" > gasToken-erc20.json
+
+# In this case, we'll configure the create rollup parameters to have a gas token
+jq --slurpfile c gasToken-erc20.json '.gasTokenAddress = $c[0].deployedTo' /opt/contract-deploy/create_rollup_parameters.json > /opt/zkevm-contracts/deployment/v2/create_rollup_parameters.json
+# shellcheck disable=SC1056,SC1072,SC1073,SC1009
+{{end}}
+
+# Deploy contracts.
+echo_ts "Deploying zkevm contracts to L1"
+
+echo_ts "Step 1: Preparing tesnet"
+npx hardhat run deployment/testnet/prepareTestnet.ts --network localhost | tee 01_prepare_testnet.out
+
+echo_ts "Step 2: Creating genesis"
+MNEMONIC="{{.l1_preallocated_mnemonic}}" npx ts-node deployment/v2/1_createGenesis.ts | tee 02_create_genesis.out
+
+echo_ts "Step 3: Deploying PolygonZKEVMDeployer"
+npx hardhat run deployment/v2/2_deployPolygonZKEVMDeployer.ts --network localhost | tee 03_zkevm_deployer.out
+
+echo_ts "Step 4: Deploying contracts"
+npx hardhat run deployment/v2/3_deployContracts.ts --network localhost | tee 04_deploy_contracts.out
+
+echo_ts "Step 5: Creating rollup"
+npx hardhat run deployment/v2/4_createRollup.ts --network localhost | tee 05_create_rollup.out
+
+# Combine contract deploy files.
+# At this point, all of the contracts /should/ have been deployed.
+# Now we can combine all of the files and put them into the general zkevm folder.
+echo_ts "Combining contract deploy files"
+mkdir -p /opt/zkevm
+cp /opt/zkevm-contracts/deployment/v2/deploy_*.json /opt/zkevm/
+cp /opt/zkevm-contracts/deployment/v2/genesis.json /opt/zkevm/
+cp /opt/zkevm-contracts/deployment/v2/create_rollup_output.json /opt/zkevm/
+cp /opt/zkevm-contracts/deployment/v2/create_rollup_parameters.json /opt/zkevm/
+popd
+
+# Combine contract deploy data.
+pushd /opt/zkevm/ || exit 1
+echo_ts "Creating combined.json"
+cp genesis.json genesis.original.json
+jq --slurpfile rollup create_rollup_output.json '. + $rollup[0]' deploy_output.json > combined.json
+
+# Add the L2 GER Proxy address in combined.json (for panoptichain).
+zkevm_global_exit_root_l2_address=$(jq -r '.genesis[] | select(.contractName == "PolygonZkEVMGlobalExitRootL2 proxy") | .address' /opt/zkevm/genesis.json)
+jq --arg a "$zkevm_global_exit_root_l2_address" '.polygonZkEVMGlobalExitRootL2Address = $a' combined.json > c.json; mv c.json combined.json
```

Comment:
I haven't seen this before but it seems redundant right? We already have `polygonZkEVMGlobalExitRootAddress` in combined.json

Reply:
Happy to chat about it.

I think it was introduced by @minhd-vu when adding panoptichain to the stack.

It looks like the address of the Polygon zkEVM Global Exit Root is not the same on L1 and L2?

```bash
$ kurtosis service exec cdk-v1 contracts-001 "cat /opt/zkevm/combined.json" | tail -n +2 | jq '{polygonZkEVMGlobalExitRootAddress, polygonZkEVMGlobalExitRootL2Address}'
{
  "polygonZkEVMGlobalExitRootAddress": "0x1f7ad7caA53e35b4f0D138dC5CBF91aC108a2674",
  "polygonZkEVMGlobalExitRootL2Address": "0xa40d5f56745a118d0906a34e69aec8c0db1cb8fa"
}
```